streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -641,6 +678,19 @@ public Serde valueSerde() {
         return parsed;
     }
 
+    @Override
+    public Long getLong(final String key) {
+        // return different default value for `commit.interval.ms` if EOS enabled
+        if (StreamsConfig.COMMIT_INTERVAL_MS_CONFIG.equals(key)
+            && eosEnabled
+            // use EOS default value only if no user config is set
+            && !originals().containsKey(StreamsConfig.COMMIT_INTERVAL_MS_CONFIG)) {
+            return EOS_DEFAULT_COMMIT_INTERVAL_MS;
```

Comment:
A really simple change would be to add a protected method to `AbstractConfig`: `processParsedConfigs`. This would give a hook for updating the parsed configs but with a clear scope (i.e. during construction only).